### ISSUE TITLE
SERVER DEATH MESSAGE CAUSES PLAYER TO BE IN DEAD STATE - SOFTLOCK (CRASH)

### ISSUE THREAD

**Issue by Camawama** [JMilamber/Brass_Amber_BattleTowers#104]
**Describe the bug**
When a player dies on the server, everything breaks. They cannot break blocks, open their inventory, and they are stuck in a "dead" state.

**To Reproduce**
Things needed to reproduce the behavior:
1. Launch an instance of my modpack 'Vanilla Plus by Cama' (you don't have to install any mods yourself).

2. Datapacks?
[Towers_Of_The_Wild_Reworked_v3.0.0_Waystones.zip](https://github.com/BrassAmber-Mods/Brass_Amber_BattleTowers/files/10008521/Towers_Of_The_Wild_Reworked_v3.0.0_Waystones.zip)

4. This only happens server-side

5. Running on Windows

**Expected behavior**
When the player dies, they should respawn normally.

**Logs**
https://pastebin.com/WSVdSQQH

**Screenshots**
![image](https://user-images.githubusercontent.com/43620900/201810673-bd1cea67-7e25-4e40-b05b-1f43e420e491.png)
Have I become a God?

**Additional context**
Help

**Comment by Camawama** [comment]
This is the modpack: https://www.curseforge.com/minecraft/modpacks/vanilla-plus-by-cama

EDIT: Server is now updated and BattleTowers was removed. In order to test this you'll have to make your own server.

**Comment by JMilamber** [comment]
Apologies for my slow response, currently on a business trip.

I believe I know why this is occurring and it is related tot the recent fix I made for another server bug.

Essentially I had for a while a class with an event onPlayerDeath that was not registered with the Forge event system. I registered it with the system when I also added events for world loading/unloading for saving BattleTower locations.

Unfortunately I had not planned ahead for the original onPlayerDeath vent to be run server side, and it contains a call to Minecraft.getInstance() which does not work server side. 

I will try to fix this in the next few days and upload a new jar by Friday.
